JoinColony/colonyCDapp
# Fix: Handle transactions that don't exist

## Description

- Properly handle the case in the action sidebar when opening an action that doesn't exist.
- Two cases are covered:
  - When the transaction hash is incorrect
  - When the transaction can't be found in the db

## Testing

- Navigate to the following URL: `http://localhost:9091/planex/activity?tx=0x`
- The action panel should load with a 404 message.

- Navigate to the following URL: `http://localhost:9091/planex/activity?tx=0xfd44e904a267f7c1377486f03020effc50139df775007be4efb7f5bf102b8a7b`
- The action panel should try to load the action for 20 seconds, before throwing a 404 (with a retry button).

## Diffs

**New stuff** ✨

* New reusable 404 message component. The 404 page now uses this too.

**Changes** 🏗

* Update the action sidebar to handle not found actions.
* Update the 404 page to use a reusable component.

## TODO

- [x] Check with @arrenv and @melyndav about the retry button for the second case, since it's different functionality from the original issue / designs. It came about from conversation with some dev team members as a nice feature to include, since an action might take some time to be created in the db, so it being missing might just be a time issue. This change can easily be removed if it's not wanted however.

Resolves #2335
